snapcraft/project/_project_info.py

```diff
@@ -45,12 +45,24 @@ def __init__(self, *, snapcraft_yaml_file_path) -> None:
         self.architectures = self.__raw_snapcraft.get("architectures")
         self.grade = self.__raw_snapcraft.get("grade")
         self.base = self.__raw_snapcraft.get("base")
+        self.build_base = self.__raw_snapcraft.get("build-base")
         self.type = self.__raw_snapcraft.get("type")
 
     def validate_raw_snapcraft(self):
         """Validate the snapcraft.yaml for this project."""
         _schema.Validator(self.__raw_snapcraft).validate()
 
+    def get_effective_base(self) -> str:
```

Comment:
With the split, maybe be explicit: e.g. get_runtime_base() vs. get_build_base()?

Reply:
I could just call this `get_build_base` which is what we really want (that's why I called it _effective_ but I can see how this can lead to confusion in the future)